src/main/java/net/minestom/server/instance/lighting/ChunkLight.java

```diff
@@ -0,0 +1,155 @@
+package net.minestom.server.instance.lighting;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.instance.Chunk;
+import net.minestom.server.network.packet.server.play.UpdateLightPacket;
+
+/**
+ * Configure chunk light data.
+ * <p>
+ * Notably used in {@link ChunkLightEngine#lightChunk(ChunkLight)} when implementing {@link ChunkLightEngine}
+ */
+public class ChunkLight {
+
+    // Data
+    final Chunk chunk;
+
+    // List of sky's light levels
+    List<byte[]> skyLight;
```

Comment:
Efficiency wise, this is incorrect.

The data should be stored as nibbles, or 4 bits each. We only need 4 bits to represent the 16 light values, and should only use that. This means that block and skylight can each be represented by an array of 2048 bytes for each chunk section.

Reply:
This class was originally meant to be per-chunk instead of per-chunk-section. The List is meant for the data in EACH chunk section, so each byte array is 2048 bytes in length, with each index representing a chunk section in a chunk.